### PR TITLE
Add pulse command and refactor

### PR DESCRIPTION
**Pull request by FoolHen**
This allows for a simpler way of doing recurring fading effects like the double vision effect, which might be fixed with this new method

fix https://github.com/BF3RM/RealityMod/issues/1491

### CODE REVIEW

File: ext/Client/__init__.lua [hunk 1]
```diff
@@ -288,32 +303,39 @@ function VEManagerClient:OnVEGuidRequest(p_ID)
...
+---@param p_ID string
 function VEManagerClient:OnVEDestroyRequest(p_ID)
 	if not self:CheckPresetID(p_ID) then return end
 
 	self.m_Presets[p_ID].ve = nil
 	self.m_Presets[p_ID].entity:Destroy()
 end
 
----@param p_ID string|nil
----@param p_Replacement string|nil
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua:300]: it is a json string

File: ext/Client/__init__.lua [hunk 2]
```diff
@@ -288,32 +303,39 @@ function VEManagerClient:OnVEGuidRequest(p_ID)
...
 			local s_Class = SkyComponentData(l_Class)
 			s_Class:MakeWritable()
-			s_Class[p_Path] = TextureAsset(ResourceManager:SearchForInstanceByGuid(p_Guid))
+
+			local s_Instance = ResourceManager:SearchForInstanceByGuid(p_Guid)
+
+			if s_Instance then
+				s_Class[p_Path] = TextureAsset(s_Instance)
+			else
+				m_Logger:Error('Could not find instance with guid ' .. tostring(p_Guid))
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: Lets not Error this. This method is used in VEEditor to change textures. Warning is enough.

**Review by IllustrisJack — CHANGES_REQUESTED** [review]

File: ext/Client/__init__.lua (hunk 1 above)
Comment (IllustrisJack) [reply ext/Client/__init__.lua:300]: (because u changed it to table)

File: ext/Client/__init__.lua [hunk 3]
```diff
@@ -121,7 +122,7 @@ function VEManagerClient:EnablePreset(p_ID)
 	end
 end
 
----@param p_ID string|nil
+---@param p_ID string
 function VEManagerClient:DisablePreset(p_ID)
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua:126]: This needs to clear self.m_Lerping[p_ID] else it will just continue pulsing

Comment (FoolHen) [reply ext/Client/__init__.lua:126]: DisablePreset calls _DestroyVE, which does that

File: ext/Client/__init__.lua (hunk 1 above)
Comment (FoolHen) [reply ext/Client/__init__.lua:300]: As far as I see m_RawPresets hold lua tables from the decoded json, so if p_Replacement is a string this is missing a json.parse

Comment (IllustrisJack) [reply ext/Client/__init__.lua:300]: oh yeah forget what I said, I decode it in veeditor

**Review by IllustrisJack — APPROVED** [review]

**Review by IllustrisJack — APPROVED** [review]

**Pull request merged**